x-pack/plugins/apm/public/components/alerting/ui_components/alert_details_app_section/alert_details_app_section.tsx

```diff
@@ -0,0 +1,464 @@
+  /**
+   * This is part or the requirements (RFC).
+   * If the alert is less than 20 units of `FOR THE LAST <x> <units>` then we should draw a time range of 20 units.
+   * IE. The user set "FOR THE LAST 5 minutes" at a minimum we should show 100 minutes.
+   */
+  const rangeFrom =
+    alertDurationMS < TWENTY_TIMES_RULE_WINDOW_MS
+      ? moment(alert.start)
+          .subtract(TWENTY_TIMES_RULE_WINDOW_MS, 'millisecond')
+          .toISOString()
+      : moment(alert.start)
+          .subtract(ruleWindowSizeMS, 'millisecond')
+          .toISOString();
+
+  const rangeTo = alert.active
+    ? 'now'
+    : moment(alert.fields[ALERT_END])
+        .add(ruleWindowSizeMS, 'millisecond')
+        .toISOString();
+
+  const { start, end } = useTimeRange({ rangeFrom, rangeTo });
+  const { agentName } = useServiceAgentFetcher({
+    serviceName,
+    start,
+    end,
+  });
+  const transactionTypes = useServiceTransactionTypesFetcher({
+    serviceName,
+    start,
+    end,
+  });
+
+  const transactionType = getTransactionType({
+    transactionType: String(alert.fields[TRANSACTION_TYPE]),
+    transactionTypes,
+    agentName,
+  });
+
+  const comparisonChartTheme = getComparisonChartTheme();
+  const INITIAL_STATE = {
+    currentPeriod: [],
+    previousPeriod: [],
+  };
+
+  /* Latency Chart */
+  const { data, status } = useFetcher(
+    (callApmApi) => {
+      if (
+        serviceName &&
+        start &&
+        end &&
+        transactionType &&
+        latencyAggregationType
+      ) {
+        return callApmApi(
+          `GET /internal/apm/services/{serviceName}/transactions/charts/latency`,
+          {
+            params: {
+              path: { serviceName },
+              query: {
+                environment,
+                kuery: '',
+                start,
+                end,
+                transactionType,
+                transactionName: undefined,
+                latencyAggregationType,
+              },
+            },
+          }
+        );
+      }
+    },
+    [
+      end,
+      environment,
+      latencyAggregationType,
+      serviceName,
+      start,
+      transactionType,
+    ]
+  );
+
+  const memoizedData = useMemo(
+    () =>
+      getLatencyChartSelector({
+        latencyChart: data,
+        latencyAggregationType,
+        previousPeriodLabel: '',
+      }),
+    // It should only update when the data has changed
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [data]
+  );
+  const { currentPeriod, previousPeriod } = memoizedData;
+
+  const timeseriesLatency = [
+    currentPeriod,
+    comparisonEnabled && isTimeComparison(offset) ? previousPeriod : undefined,
+  ].filter(filterNil);
+
+  const latencyMaxY = getMaxY(timeseriesLatency);
+  const latencyFormatter = getDurationFormatter(latencyMaxY);
+  const preferredAnomalyTimeseriesLatency =
+    usePreferredServiceAnomalyTimeseries(ApmMlDetectorType.txLatency);
```

Comment:
This doesn't work without the appropriate context. I don't think we need to use it here, or we need to wrap this component with the context providers.

Reply:
I didn't know that; I was aiming to keep the style of charts aligned on both pages(APM and Observability). 
After deleting them, the style is still the same, which is perfect!